.github/workflows/cache-eviction.yaml

```
@@ -0,0 +1,35 @@
+name: Cache Eviction
+
+on:
+  pull_request:
+    types: closed
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  evict-mod-cache-closed-pr:
+    name: Evict Go Modules Cache for Closed PRs
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Cleanup Cache for PR
+        run: |
+          gh extension install actions/gh-actions-cache
+
+          REPO=${{ github.repository }}
+          BRANCH="refs/pull/${{ github.event.pull_request.number }}/merge"
+
+          echo "Fetching list of cache keys for branch $BRANCH (max 100)"
```

```suggestion
          echo "Fetching list of cache keys for branch $BRANCH (there can be at most 100 keys per branch)"
```

Related to the below conversation, I'm hoping to make it clearer that this number is derived from san external constraint, not something we're deciding on for this job in particular
As written could be read simply as an explanation of the following like where we get 100 lines
I'm open to better wording